### DISCUSSION TITLE
weighted-search | Compound Queries with Weighted Fields

### DISCUSSION THREAD

**Discussion by MwSpaceLLC** [prisma/prisma#16852] [Q&A]
Hi there, today I'm going to create with my teame query for order by case as follow:

```sql
SELECT ragione_sociale, localita, nazione, macrocategoria, microcategoria,
IF(
`ragione_sociale` LIKE "bar%",  20,
IF(`ragione_sociale` LIKE "%bar%", 10, 0)
)
+ IF(`macrocategoria` LIKE "%bar%", 5,  0)
+ IF(`microcategoria`         LIKE "%bar%", 1,  0)
AS `weight`
FROM anagrafiche
WHERE (
    `ragione_sociale` LIKE "%bar%"
    OR `macrocategoria` LIKE "%bar%"
    OR `microcategoria`         LIKE "%bar%"
)
ORDER BY weight DESC
```

OR:

```sql
SELECT country_code,ragione_sociale, localita, nazione, macrocategoria, microcategoria
FROM anagrafiche
WHERE (
              `ragione_sociale` LIKE "%bar%"
              OR `macrocategoria` LIKE "%bar%"
              OR `microcategoria` LIKE "%bar%"
          )
ORDER BY CASE
             WHEN `ragione_sociale` LIKE "% bar %" THEN ragione_sociale
             WHEN `macrocategoria` LIKE "%bar%" THEN macrocategoria
             WHEN `microcategoria` LIKE "%bar%" THEN microcategoria
             END DESC
```

OR:

```sql
SELECT ragione_sociale,
       localita,
       nazione,
       macrocategoria,
       microcategoria,
       IF(MATCH (ragione_sociale) AGAINST ('bar' IN NATURAL LANGUAGE MODE ), 20, 0)
           + IF(MATCH (macrocategoria) AGAINST ('bar' IN NATURAL LANGUAGE MODE), 5, 0)
           + IF(MATCH (microcategoria) AGAINST ('bar' IN NATURAL LANGUAGE MODE), 1, 0)
           AS `weight`
FROM anagrafiche
WHERE country_code = 'IT'
  AND (
        MATCH (ragione_sociale) AGAINST ('bar' IN NATURAL LANGUAGE MODE)
        OR MATCH (macrocategoria) AGAINST ('bar' IN NATURAL LANGUAGE MODE)
        OR MATCH (microcategoria) AGAINST ('bar' IN NATURAL LANGUAGE MODE)
    )
ORDER BY weight DESC LIMIT 100
```

 We want use prisma modelling but we can't find the correct method in the guide.

Also fulltext mysql already support search score:

`SELECT * FROM anagrafiche WHERE MATCH(ragione_sociale, macrocategoria, microcategoria) AGAINST('bar w/2 ragione_sociale:3 macrocategoria:1 microcategoria:1');`

The aspected records order depends by result in query (match like fulltextindex)

Tnks for any reply 🤖

**Comment by MwSpaceLLC** [comment]
[MongoDB Atlas](https://www.mongodb.com/docs/atlas/)

Use custom scoring to alter the relevance score returned is very important for many use case

**Reply by nurul3101** [reply]
It looks like this is a duplicate of GitHub Discussion: https://github.com/prisma/prisma/discussions/16222

**Comment by MwSpaceLLC** [comment]
Please, check also pgsql score. THIS SUPPORT IS NO OPTIONAL, REQUIRED IN MOST COMPANIES

https://www.postgresql.org/docs/current/textsearch-controls.html

**Reply by nurul3101** [reply]
We have a related feature request here: [#5027](https://github.com/prisma/prisma/issues/5027), if you can leave a 👍 to the feature request then it would help our product team in prioritising it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
